x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/index.ts

```diff
@@ -79,10 +80,17 @@ export function createApmEventClient({
   return {
     async search<TParams extends APMEventESSearchRequest>(
       params: TParams,
-      { includeLegacyData = false } = {}
+      optionsOrOperationName:
+        | string
+        | { includeLegacyData?: boolean; operationName: string }
     ): Promise<TypedSearchResponse<TParams>> {
       const withProcessorEventFilter = unpackProcessorEvents(params, indices);
 
+      const { includeLegacyData = false, operationName } =
+        typeof optionsOrOperationName === 'string'
+          ? { operationName: optionsOrOperationName }
+          : optionsOrOperationName ?? {};
```

Comment:
Can you also add operation name to the debug output? Will make it so much easier to find the relevant query